packages/components/addon/components/hds/stepper/indicator/task.js

```diff
@@ -35,6 +34,29 @@ export default class HdsStepperIndicatorTaskIndexComponent extends Component {
     return this.args.isInteractive || false;
   }
 
+  /**
+   * @param iconName
+   * @type {string}
+   */
+
+  get iconName() {
+    let status = this.args.status;
+
+    // Check the status using a ternary operator and returning a icon name value
+    const value =
+      status === 'incomplete'
+        ? 'circle'
+        : status === 'progress'
+        ? 'circle-half'
+        : status === 'processing'
+        ? 'loading'
+        : status === 'complete'
+        ? 'check-circle'
+        : null;
```

Comment:
maybe you could use a mapping object at the top, and then return the value given the key? see for example how it's done here: https://github.com/hashicorp/design-system/blob/580c8ddf765fbd3fff03f42868c1f696f72df9cd/packages/components/addon/components/hds/alert/index.js#L9-L15

I mean something like this:

```
const MAPPING_STATUS_TO_ICON = {
  'incomplete': 'circle',
  'progress': 'circle-half',
  'processing': 'loading',
  'complete': 'check-circle',
};

...

get iconName() {
  return MAPPING_STATUS_TO_ICON(this.status);
}      
```

Reply:
Great, exactly what I was looking for. I was wondering whether a map would work better rather than a cascading ternary statement.

Reply:
> whether a map would work better rather than a cascading ternary statement.

Are you _really_ asking this? 🤔

(just kidding 🤣)